dbt/include/global_project/macros/adapters/relation.sql

```diff
@@ -7,6 +7,11 @@
 {% endmacro %}
 
 {% macro make_temp_relation(base_relation, suffix='__dbt_tmp') %}
+  {#-- This ensures microbatch batches get unique temp relations to avoid clobbering --#}
+  {% if suffix == '__dbt_tmp' and model.batch %}
```

Comment:
Is there a reason to limit this to a particular suffix? Or can we just look for the existence of a batch configuration?

Reply:
My thought process was that if a specific suffix is being passed in, we probably don't want to mess with it, as we don't know if doing so is safe or not.

Reply:
If we don't limit to the `__dbt_tmp` suffix, it won't be possible for users to provide their own custom suffix for microbatch models. It's unclear whether that's actually desirable or not. If we're saying that microbatch temp tables should always be suffixed with a batch-specific id, then we shouldn't restrict to a particular suffix. If we want to give users the flexibility to fully manage their own temp table names, even in microbatch, then Quigley's current solution is more appropriate.

Reply:
> If we want to give users the flexibility to fully manage their own temp table names, even in microbatch, then Quigley's current solution is more appropriate.

This is the distinction I'm looking for. If that's the intention, and we don't need to do this for other built-in suffixes (e.g. `__dbt_backup`), then this is fine. I suspect we may want to broaden this in the future to some type of "dbt-managed" versus "non-dbt-managed", but that's well outside of the scope of this PR.